src/Geno/C++/GUI/Widgets/TextEdit.cpp

```diff
@@ -3439,3 +3507,590 @@ std::vector< int > TextEdit::CursorsNotInText( File& rFile )
 
 	return Cursors;
 } // CursorsNotInText
+
+//////////////////////////////////////////////////////////////////////////
+
+void TextEdit::ClearSearch( File& rFile )
+{
+	delete rFile.SearchDiag.SearchResult;
+	rFile.SearchDiag.SearchResult = nullptr;
+
+	StatusBar::Instance().SetSearchResultInfo("", 0);
+} // ClearSearch
+
+//////////////////////////////////////////////////////////////////////////
+
+void TextEdit::PrepareSearchString( std::string& rSearchString )
+{
+	for( int i = 0; i < ( int )rSearchString.length(); i++ )
+	{
+		char C = rSearchString[ i ];
+
+		if( C == '\\' )
+		{
+			if( i == ( int )rSearchString.length() - 1 ) break;
+
+			char C2 = rSearchString[ i + 1 ];
+
+			switch( C2 )
+			{
+				case 'n':
+					rSearchString[ i ] = '\n';
+					break;
+				case 't':
+					rSearchString[ i ] = '\t';
+					break;
+				default:
+					continue;
+			}
+
+			rSearchString.erase( rSearchString.begin() + i + 1 );
+		}
+	}
+} // PrepareSearchString
```

Comment:
What if we are searching for a backslash followed by an 'n' character, and not a newline? We can't type double backslashes to escape it. Should be always escape the following character after a backslash maybe?